src/rgw/rgw_mdlog.h

```diff
@@ -79,6 +79,39 @@ class RGWMetadataLogInfoCompletion : public RefCountedObject {
   }
 };
 
+class RGWMetadataLogBE {
+protected:
+  CephContext* const cct;
+private:
+  std::string prefix;
+ 
+public:
+  using entries = std::variant<std::list<cls_log_entry>,
+			       std::vector<ceph::buffer::list>>;
+
+  RGWMetadataLogBE(CephContext* const cct, std::string prefix);
+  virtual ~RGWMetadataLogBE();
+
+  std::string get_shard_oid(int id) const {
+    return fmt::format("{}{}", prefix, id);
+  }
+
+  virtual int push(int index, std::list<cls_log_entry>& items, librados::AioCompletion *&completion) { return 0; };
+  virtual int push(int index, ceph::real_time now,
+		   const std::string& section,
+		   const std::string& key,
+		   ceph::buffer::list& bl) { return 0; };
+  virtual int list(int index, int max_entries,
+		   std::list<cls_log_entry>& items,
```

Comment:
If possible, I'd rather take the opportunity to swap out list for vector.

Reply:
yes yes yes!!!  also look daggers at std::maps flying around, if you see them

Reply:
okay.